diff --git a/dist/cjs/internal/commandDescriptor.js b/dist/cjs/internal/commandDescriptor.js
index 943ee7c48c59412ecb5f2eb2d09f58e1c4d9e084..83e6280042576fed182505975582781c7100ad72 100644
--- a/dist/cjs/internal/commandDescriptor.js
+++ b/dist/cjs/internal/commandDescriptor.js
@@ -175,10 +175,11 @@ const getHelpInternal = (self, config) => {
             case "Subcommands":
               {
                 const parentUsage = getUsage(command.parent, preceding);
+                const parentSelfUsage = InternalHelpDoc.getSpan(InternalUsage.getHelp(getUsageInternal(command.parent)));
                 return Option.match(Arr.head(parentUsage), {
                   onNone: () => Arr.flatMap(command.children, child => getUsage(child, preceding)),
-                  onSome: ([usage]) => {
-                    const childrenUsage = Arr.flatMap(command.children, child => getUsage(child, Arr.append(preceding, usage)));
+                  onSome: () => {
+                    const childrenUsage = Arr.flatMap(command.children, child => getUsage(child, Arr.append(preceding, parentSelfUsage)));
                     return Arr.appendAll(parentUsage, childrenUsage);
                   }
                 });
diff --git a/dist/esm/internal/commandDescriptor.js b/dist/esm/internal/commandDescriptor.js
index d5deeae817197cf572fd2dd4aaf539aa759c80ce..aac24137d9ff85332aab564aefa76be8f2db7e69 100644
--- a/dist/esm/internal/commandDescriptor.js
+++ b/dist/esm/internal/commandDescriptor.js
@@ -154,10 +154,11 @@ const getHelpInternal = (self, config) => {
             case "Subcommands":
               {
                 const parentUsage = getUsage(command.parent, preceding);
+                const parentSelfUsage = InternalHelpDoc.getSpan(InternalUsage.getHelp(getUsageInternal(command.parent)));
                 return Option.match(Arr.head(parentUsage), {
                   onNone: () => Arr.flatMap(command.children, child => getUsage(child, preceding)),
-                  onSome: ([usage]) => {
-                    const childrenUsage = Arr.flatMap(command.children, child => getUsage(child, Arr.append(preceding, usage)));
+                  onSome: () => {
+                    const childrenUsage = Arr.flatMap(command.children, child => getUsage(child, Arr.append(preceding, parentSelfUsage)));
                     return Arr.appendAll(parentUsage, childrenUsage);
                   }
                 });
